app/presenters/queries/link_set_presenter.rb

```diff
@@ -22,7 +25,21 @@ def present
       end
 
       def links
-        present[:links]
+        # This method presents LinkSet#links as a hash.
+        # Where keys are link types and their values are arrays of target content ids.
+        # ie:
+        # {
+        #   related: [ UUID, UUID ],
+        #   organisations: [ UUID ],
+        # }
+
+        link_set.links.map.with_object({}) do |link, hash|
+          if hash.has_key?(link.link_type)
+            hash[link.link_type] << link.target_content_id
+          else
+            hash[link.link_type] = [link.target_content_id]
+          end
+        end.symbolize_keys
       end
```

Comment:
Something link this might be a bit more readable:

```ruby
link_set.links.each.with_object({}) do |link, hash|
  type = link.link_type.to_sym

  hash[type] ||= []
  hash[type] << link.target_content_id
end
```

(I haven't tested this)